crates/eip4844/src/prover.rs

```diff
@@ -0,0 +1,238 @@
+use bls12_381::{batch_inversion::batch_inverse, ff::Field, lincomb::g1_lincomb, G1Point, Scalar};
+use maybe_rayon::prelude::*;
+use polynomial::domain::Domain;
+
+use crate::{
+    constants::FIELD_ELEMENTS_PER_BLOB,
+    serialization::{
+        deserialize_blob_to_scalars, deserialize_bytes_to_scalar, deserialize_compressed_g1,
+        serialize_g1_compressed,
+    },
+    trusted_setup::{deserialize_g1_points, SubgroupCheck},
+    verifier::{bitreverse, bitreverse_slice, compute_fiat_shamir_challenge},
+    BlobRef, Context, Error, KZGCommitment, KZGOpeningEvaluation, KZGOpeningPoint, KZGProof,
+    TrustedSetup,
+};
+
+/// The key that is used to commit to polynomials in lagrange form (bit-reversed
+/// order)
+pub struct CommitKey {
+    g1_lagrange_brp: Vec<G1Point>,
+}
+
+impl From<&TrustedSetup> for CommitKey {
+    fn from(setup: &TrustedSetup) -> Self {
+        let mut g1_lagrange_brp = deserialize_g1_points(&setup.g1_lagrange, SubgroupCheck::NoCheck);
+        bitreverse_slice(&mut g1_lagrange_brp);
+        Self { g1_lagrange_brp }
+    }
+}
```

Comment:
So that we can eventually share more code with eip7594 -- we can leave the commitKey as it is and bit_reverse the polynomial each time, before we commit to it in `blob_to_kzg_commitment`

The time to bit_reverse the polynomial is negligible given we need to do a `g1_lincomb`